### PR TITLE
f_lavfi: don't reject dynamic lavfi ins/outs (v2)

### PR DESCRIPTION
**Pull request by llyyr**
Ideally, users should be using lavfi-complex instead of lavfi-bridge for such a use case, however currently lavfi-complex doesn't support hwdec. So we can allow filters with dynamic inputs to work with lavfi-bridge, at the cost of them only being able to take in only one input. This should probably be reverted when/if lavfi-complex has hwdec, but for now this allows us to use libplacebo as a video filter with hwdec in mpv again.

Basically the same as #12170 but Github don't let you reopen PRs if you force pushed while it was closed...

I've made the code changes requested by @kasper93, and also added a note about --vf limitations in the docs. I'm not particularly sold on the idea of allowing filters with dynamic in/out in lavfi-bridge, but there are certainly limitations with lavfi-complex that aren't super simple to resolve, so this may have some utility.

### CODE REVIEW

**Comment by kasper93** [comment]
LGTM

**Comment by Dudemanguy** [comment]
If the problem is just "I want to use hardware decoding with lavfi-complex", then I'm of the opinion lavfi-complex should be made smarter like what was done in https://github.com/mpv-player/mpv/commit/989d873d6ec57171a55f432d6f87a9e5a61a706c (evidently this didn't work for vf until late last year).

**Comment by llyyr** [comment]
Well, modifying the graph is also more convenient with --vf. I don't really understand what the use case for this is, but they may be better served using ffmpeg directly. As long as nobody complains about any bugs with this, I don't see the problem allowing it for now though

**Comment by hooke007** [comment]
haasn once fixed the bug mpv cannot use vf_libplacebo. Now it is broken. I cannot understand why it is reasonable to be ignored.

**Review by haasn — APPROVED** [review]
